internal/rbd/rbd_util.go

```diff
@@ -426,9 +426,24 @@ func flattenClonedRbdImages(ctx context.Context, snaps []snapshotInfo, pool, mon
 		util.ErrorLog(ctx, "failed to open connection %s; err %v", rv, err)
 		return err
 	}
-	for _, s := range snaps {
-		if s.Namespace.Type == "trash" {
-			rv.RbdImageName = s.Namespace.OriginalName
+
+	for _, snapInfo := range snaps {
+		// check if the snapshot belongs to trash namespace.
+		isTrash, err := rv.isTrashSnap(snapInfo.Id)
```

Comment:
@agarwal-mudit I think this won't be valid as the rbd image name will be updated at line 446 isn't it?

Reply:
Nice catch @Madhu-1, Yeah that should be an issue here. Let me think of a solution. Thanks for the help.